src/veranda/raster/mosaic/netcdf.py

```diff
@@ -0,0 +1,595 @@
+        unlimited_dims = to_list(unlimited_dims)
+        data_variables = list(data.data_vars)
+        all_dims = list(data.dims)
+        space_dims = all_dims[-2:]
+        stack_dim_names = all_dims[:-2]
+        stack_dims = dict()
+        for stack_dim_name in stack_dim_names:
+            if stack_dim_name in unlimited_dims:
+                stack_dims[stack_dim_name] = None
+            else:
+                stack_dims[stack_dim_name] = len(data[stack_dim_name])
+
+        nodatavals = dict()
+        scale_factors = dict()
+        offsets = dict()
+        dtypes = dict()
+        for data_variable in data_variables:
+            dtypes[data_variable] = data[data_variable].data.dtype.name
+            nodatavals[data_variable] = data[data_variable].attrs.get('_FillValue', 0)
+            scale_factors[data_variable] = data[data_variable].attrs.get('scale_factor', 1)
+            offsets[data_variable] = data[data_variable].attrs.get('add_offset', 0)
+
+        for filepath, file_group in self._file_register.groupby('filepath'):
+            tile_id = file_group.iloc[0].get('tile_id', '0')
+            file_coords = list(file_group[self._file_dim])
+            tile = self._mosaic[tile_id]
+            if not tile.intersects(data_geom):
+                continue
+            file_id = file_group.iloc[0].get('file_id', None)
+            if file_id is None:
+                gt_driver = NetCdf4File(filepath, mode='w', geotrans=tile.geotrans, sref_wkt=tile.sref.wkt,
+                                        stack_dims=stack_dims,
+                                        space_dims={space_dims[0]: tile.n_rows, space_dims[1]: tile.n_cols},
+                                        data_variables=data_variables, dtypes=dtypes,
+                                        scale_factors=scale_factors, offsets=offsets, nodatavals=nodatavals,
+                                        attrs={'time': {'units': 'days since 1950-01-01 00:00:00'}},  # TODO: make this more flexible (this needs to be defined from outside)
+                                        metadata=data.attrs, **kwargs)
+                file_id = len(list(self._files.keys())) + 1
+                self._files[file_id] = gt_driver
+                self._file_register.loc[file_group.index, 'file_id'] = file_id
+
+            nc_file = self._files[file_id]
+            tile_write = data_geom.slice_by_geom(tile, inplace=False)
+            raster_access = RasterAccess(tile_write, tile, src_root_raster_geom=data_geom)
+            xrds = data.sel(**{self._file_dim: file_coords,
+                               space_dims[0]: tile_write.y_coords,
+                               space_dims[1]: tile_write.x_coords})
+            data_write = xrds[data_variables]
+            nc_file.write(data_write, row=raster_access.dst_window[0], col=raster_access.dst_window[1],
+                          encoder=encoder, encoder_kwargs=encoder_kwargs)
```

Comment:
same goes for this section. see if you can not reuse logic from the geotiff version and put into a common paramterized function

Reply:
Done.